pkg/controller/registry/reconciler/configmap.go

```diff
@@ -513,5 +515,36 @@ func (c *ConfigMapRegistryReconciler) CheckRegistryServer(logger *logrus.Entry,
 	}
 
 	healthy = true
+	if deadPodsDetected, e := detectAndDeleteDeadPods(logger, c.OpClient, pods, source.GetNamespace()); deadPodsDetected {
+		healthy = false
+		err = e
+	}
+	return
+}
+
+// detectAndDeleteDeadPods determines if there are registry client pods that are in the deleted state
+// but have not been removed by GC (eg the node goes down before GC can remove them), and attempts to
+// force delete the pods
+func detectAndDeleteDeadPods(logger *logrus.Entry, client operatorclient.ClientInterface, pods []*corev1.Pod, sourceNamespace string) (foundDeadPod bool, err error) {
```

Comment:
Nit (but sorta major nit): I know there are other places in the old OLM code that use named/naked returns, but these almost always hurt code readability. Not suggesting we go and fix all of those, but I think it would be good to avoid adding more.

WDYT about using `(bool, error)` in the signature and have explicit returns?

Reply:
Sure I can do that, I don't like named returns either, just went with it since `CheckRegistryServer` implementation for ConfigMap already has that. But I'll update the PR with just `(bool, error)` in my next update...

Reply:
Done.